openmmtools/mcmc.py

```diff
@@ -234,11 +290,16 @@ class MCMCSampler(object):
 
     """
 
-    def __init__(self, thermodynamic_state, sampler_state, move):
+    def __init__(self, thermodynamic_state, sampler_state, move,
+                 energy_context_cache=None, integrator_context_cache=None):
```

Comment:
The `MCMCSampler` isn't actually used in `MultiStateSampler`, and doesn't need separate `_energy_context_cache` or `_propagation_context_cache` attributes. 

In fact, we don't need to store any context caches at all---we can just accept an optional `context_cache` argument in `run()`, where we would default to `cache.global_context_cache` if none is specified.

Reply:
As discussed we now have `energy_context_cache` and `sampler_context_cache` as attributes that are passed in `create()` and default to using the global context cache if `None`.

Reply:
Ok, I'll make the changes, this one is still missing in the current status of the PR.

Reply:
Changed in current state of the PR.